apps/web/src/app/(signing)/sign/[token]/page.tsx

```diff
@@ -53,6 +53,13 @@ export default async function SigningPage({ params: { token } }: SigningPageProp
 
   const user = await getServerComponentSession();
 
+  if (
+    document.status === DocumentStatus.COMPLETED ||
+    recipient.signingStatus === SigningStatus.SIGNED
+  ) {
+    redirect(`/sign/${token}/complete`);
+  }
+
```

Comment:
I think this should go directly after 

```
  if (!document || !document.documentData || !recipient) {
    return notFound();
  }
```

Since it doesn't need to rely on the two sequential awaits that aren't relevant to the statement